freeipa/src/main/java/com/sequenceiq/freeipa/flow/stack/upgrade/ccm/UpgradeCcmService.java

```diff
@@ -222,7 +222,7 @@ public void failedState(UpgradeCcmContext context, UpgradeCcmFailureEvent payloa
             LOGGER.info("Re-registering cluster proxy to previous tunnel {}", payload.getOldTunnel());
             registerClusterProxy(payload.getResourceId());
         }
-        String statusReason = "Upgrade CCM failed";
+        String statusReason = "Upgrade CCM failed: " + payload.getException().getMessage();
```

Comment:
Same sanity check like above. :smile:

Reply:
No because the exception in the payload is propagated from the `catch` clause (of wherever the exception happened).